pkg/extensions/sync/on_demand.go

```diff
@@ -38,8 +47,21 @@ func NewOnDemand(log log.Logger) *BaseOnDemand {
 	return &BaseOnDemand{log: log, requestStore: &sync.Map{}}
 }
 
-func (onDemand *BaseOnDemand) Add(service Service) {
+func (onDemand *BaseOnDemand) Add(service Service, syncTimeout time.Duration) {
+	serviceID := len(onDemand.services)
+
+	// Use default if timeout not configured
+	if syncTimeout == 0 {
+		syncTimeout = syncOperationTimeout
+	}
+
+	onDemand.log.Info().
```

Comment:
These were helpful to me during debugging. I am happy to remove them again if you prefer.

Reply:
If this is only for debugging, move this to Debug() instead.